pandas/tests/indexes/multi/test_drop.py

```diff
@@ -139,3 +139,32 @@ def test_drop_not_lexsorted():
     tm.assert_index_equal(lexsorted_mi, not_lexsorted_mi)
     with tm.assert_produces_warning(PerformanceWarning):
         tm.assert_index_equal(lexsorted_mi.drop("a"), not_lexsorted_mi.drop("a"))
+
+
+def test_drop_with_non_unique_datetime_index_and_invalid_keys():
+    # GH 30399
+
+    # define dataframe with unique datetime index
+    df_unique = pd.DataFrame(
+        np.random.randn(5, 3),
+        columns=["a", "b", "c"],
+        index=pd.date_range("2012", freq="H", periods=5),
+    )
+    # create dataframe with non-unique datetime index
+    df_nonunique = df_unique.copy().iloc[[0, 2, 2, 3]]
+
+    try:
```

Comment:
I think the point of this is the raise a `KeyError` for the example specified - can you model the test to use `pytest.raises` instead? Should see other examples of this in the code base

Reply:
Have updated the code as requested. Now the test specifically expects KeyError with string "not found in axis" within the message